evm/src/ext.rs

```diff
@@ -0,0 +1,28 @@
+use ethereum_types::*;
+
+pub trait DataProvider {
+    fn get_balance(&self, address: &Address) -> U256;
+
+    fn add_refund(&mut self, n: u64);
+    fn sub_refund(&mut self, n: u64);
+    fn get_refund(&self) -> u64;
+
+    fn get_code_size(&self, address: &Address) -> u64;
+    fn get_code(&self, address: &Address) -> &[u8];
+    fn get_code_hash(&self, address: &Address) -> H256;
+
+    fn get_block_hash(&self, number: &U256) -> H256;
+
+    fn get_storage(&self, address: &Address, key: &H256) -> H256;
```

Comment:
How about returns `Result<U256>`?

Reply:
Same as returns H256::zero() if the key not exists.

Reply:
What about database error?  EVM doesn't seem to handle it.....

Reply:
Even if you give me an error, I will continue to execute as H256::zero(); The interpreter is no way to handle the db error. 
The wrong block will not be accepted by other nodes. So I think it's OK.